staking/effective/calculate.go

```diff
@@ -148,13 +145,24 @@ func Compute(
 }
 
 // Apply ..
-func Apply(shortHand map[common.Address]*SlotOrder, pull int) (
+func Apply(shortHand map[common.Address]*SlotOrder, pull int, isExtendedBound bool) (
 	numeric.Dec, []SlotPurchase,
 ) {
 	median, picks := Compute(shortHand, pull)
+	max := onePlusC.Mul(median)
+	min := oneMinusC.Mul(median)
+	if isExtendedBound {
+		max = onePlusCV2.Mul(median)
+		min = oneMinusCV2.Mul(median)
+	}
 	for i := range picks {
-		picks[i].EPoSStake = effectiveStake(median, picks[i].RawStake)
+		picks[i].EPoSStake = effectiveStake(min, max, picks[i].RawStake)
 	}
 
 	return median, picks
 }
+
+func effectiveStake(min, max, actual numeric.Dec) numeric.Dec {
+	newMax := numeric.MinDec(max, actual)
+	return numeric.MaxDec(newMax, min)
+}
```

Comment:
I noticed that the function was just refactored, the algorithm didn't actually change, just the max and min calculations were extracted externally, right? This does make it clearer

Reply:
yes correct.